cmd/clusterctl/client/repository/template.go

```diff
@@ -30,10 +30,15 @@ import (
 // 1. Checks for all the variables in the cluster template YAML file and replace with corresponding config values
 // 2. Ensure all the cluster objects are deployed in the target namespace.
 type Template interface {
-	// Variables required by the template.
-	// This value is derived by the template YAML.
+	// Variables used by the template.
+	// This value is derived from the template YAML.
 	Variables() []string
 
+	// Variables used by the template with their default values. If the value is empty, there is no
+	// default and the variable is required.
+	// This value is derived from the template YAML.
+	VariableMap() map[string]string
```

Comment:
```suggestion
	// Variables used by the template with their default values. If the value is empty, there is no
	// default and the variable is required.
	// This value is derived from the template YAML.
	VariableMap() map[string]*string
```

Should this be a pointer instead? Or how do we know if a variable has an empty string as default or not?

Reply:
If the string is empty, it's considered required. If there was a an empty default value in the YAML, the string will contain `""` (double quote chars).

Yes, probably a string pointer would be a more canonical way to represent this, but I think I was getting stuck when trying that refactoring, so I ended up with this. Happy to revisit that if `*string` would be more legit.

Reply:
I changed it to use a `map[string]*string` because the type gives more information and as you pointed out, it's less surprising.